components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/EndpointResource.java

```diff
@@ -125,39 +129,176 @@ private JSONObject getEndpointByName(MessageContext messageContext, String endpo
 
         SynapseConfiguration configuration = messageContext.getConfiguration();
         Endpoint ep = configuration.getEndpoint(endpointName);
-        return convertEndpointToJsonObject(ep);
+        return getEndpointAsJson(ep);
     }
 
-    private JSONObject convertEndpointToJsonObject(Endpoint endpoint) {
+    /**
+     * Returns the json representation of the endpoint based on its type.
+     *
+     * @param endpoint endpoint
+     * @return json-object with endpoint details
+     */
+    private JSONObject getEndpointAsJson(Endpoint endpoint) {
 
-        if (Objects.isNull(endpoint)) {
-            return null;
+        JSONObject endpointObject = new JSONObject();
+        endpointObject.put(Constants.NAME, endpoint.getName());
+        OMElement synapseConfiguration = EndpointSerializer.getElementFromEndpoint(endpoint);
+        endpointObject.put(Constants.SYNAPSE_CONFIGURATION, synapseConfiguration);
+
+        if (endpoint instanceof AddressEndpoint) {
```

Comment:
Shall we refactor this as discussed offline?

Reply:
done